tests/drivers/video/ctrl/src/main.c

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The Zephyr Project Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/video.h>
+
+ZTEST(video_ctrl, test_video_convert_ctrl_name)
+{
+	char dest[40];
+	const char *examples[][2] = {
+		{"White Balance, Automatic", "white_balance_automatic"},
+		{"nothing exceptional", "nothing_exceptional"},
+		{"UPPERCASE", "uppercase"},
+		{"._!!_###_!!_.", ""},
+		{"  _  leading special characters", "leading_special_characters"},
+		{"trailing special characters!@#$%^", "trailing_special_characters_"},
```

Comment:
so trailing special characters are folded into 1 underscore and leading special characters end up being discarded? the lack of symmetry seems odd, maybe?

Reply:
> the lack of symmetry seems odd, maybe?

I agree. This was assuming a control would never be named i.e. `Misc +`.
Latest push also covers this edge case instead of asserting a mis-handling.